traces/stf_trace_gen/dromajo_stf_lib.patch

```diff
@@ -81,212 +213,107 @@ index 2f71c99..2de97a3 100644
-+        }
 +
-+        if(tracing_enabled)
-+        {
-+            // Only trace in user priv and the same application that
-+            // started the trace
-+            if((priv == 0) &&
-+               (cpu->pending_exception == -1) &&
-+               (stf_prog_asid == ((cpu->satp >> 4) & 0xFFFF)))
-+            {
-+                ++stf_count;
-+                const uint32_t inst_width = ((insn_raw & 0x3) == 0x3) ? 4 : 2;
-+                bool skip_record = false;
-+
-+                // See if the instruction changed control flow or a
-+                // possible not-taken branch conditional
-+                if(cpu->info != ctf_nop) {
-+                    stf_writer << stf::InstPCTargetRecord(virt_machine_get_pc(m, 0));
-+                }
-+                else {
-+                    // Not sure what's going on, but there's a
-+                    // possibility that the current instruction will
-+                    // cause a page fault or a timer interrupt or
-+                    // process switch so the next instruction might
-+                    // not be on the program's path
-+                    if(cpu->pc != last_pc + inst_width) {
-+                        skip_record = true;
-+                    }
-+                }
-+
-+                // Record the instruction trace record
-+                if(false == skip_record)
-+                {
-+                    // If the last instruction were a load/store,
-+                    // record the last vaddr, size, and if it were a
-+                    // read or write.
-+                    if(cpu->last_data_vaddr != std::numeric_limits<decltype(cpu->last_data_vaddr)>::max())
-+                    {
-+                        stf_writer << stf::InstMemAccessRecord(cpu->last_data_vaddr,
-+                                                               cpu->last_data_size,
-+                                                               0,
-+                                                               (cpu->last_data_type == 0) ?
-+                                                               stf::INST_MEM_ACCESS::READ :
-+                                                               stf::INST_MEM_ACCESS::WRITE);
-+                        stf_writer << stf::InstMemContentRecord(0); // empty content for now
-+                    }
-+
-+                    if(inst_width == 4) {
-+                        stf_writer << stf::InstOpcode32Record(insn_raw);
-+                    }
-+                    else {
-+                        stf_writer << stf::InstOpcode16Record(insn_raw & 0xFFFF);
-+                    }
-+                }
-+            }
-+        }
-+        return keep_going;
-+    }
-+
-     if (!do_trace) {
-         return keep_going;
-     }
-@@ -223,7 +322,7 @@ int main(int argc, char **argv) {
-     if (!m)
-         return 1;
- 
--    int n_cycles = 10000;
-+    int n_cycles = 1;
```

Comment:
Can this be changed back to 10k?

Reply:
yes it can
mistake in the patch, my source has 10K